### ISSUE TITLE
Any way to force "thetvdb movie id"? If not, any way to force TheMovieDB ID? Documentation says deprecated.

### ISSUE THREAD

**Issue by RainOne** [ZeroQI/Hama.bundle#447]
Hi!

I've been using ASS+HAMA for a while, very successfully to index my anime collection. Since it allows for TVs and Movies to be indexed together, I thought about using it to index my Tokusatsu collection - not present on AniDB, but presented on TheTVDB and  The Movie Database.

This is entirely unsupported, and I'm aware - Tokusatsu indexing is terrible on TheTVDB (everything is under a 40+ season for Super Sentai, for instance, with some shows having their unique database entry), so manually forcing matches with "tvdb" on the folders seem to be the most sure way of properly matching everything.

The thing is that TheTVDB now has a "movies" entry - for instance, on https://thetvdb.com/movies/shin-kamen-rider-prologue
Reading ASS documentation, it seems I can no longer force TheMovieDatabase IDs with the "tmdb-" tag. Not only that, I can't find a way to force tag a "THETVDB.COM MOVIE ID" as it's listed on the website - just putting that id finds the TV ID instead.

I know about the "complete movie" workaround, but manually forcing the ID works well for me on the Anime collection and I was thinking I could do something like that on the Tokusatsu library, specially since most of the items i need to match are present on TheTVDB. Am I missing something or is there no current way to force TheTVDB.com movie ids? Is TheMovie Database forced matching also no longer possible?

Thank you in advance.

**Comment by ZeroQI** [comment]
Let's try to summarize current situation
- I will not add TheTVDB movie ID support as too much work is required 
- I am trying to add support for a Forced TheMovieDB ID to gather metadata, as ASS should be already compatible, and that is very little code since the support for TMDB ID from the mapping file is already there, just adding forced id support
- metadata showed no ID when it should have previously in logs with forced TMDB ID:```
=== TheMovieDb.GetMetadata() ================================================================================================================================
TVDBid: '', TMDbid: '', IMDbid: ''
```
- it can't load json poster, so let's put that asside for now

The action plan at this point is:
- to edit AnimeLists.py, by updating line 235 to:
return AnimeLists_dict, AniDB_winner, TVDB_winner if TVDB_winner.isdigit() else "", Dict(mappingList, 'tmdbid') or TMDBid, Dict(mappingList, 'imdbid') or IMDBid, mappingList
- Then remove lines 79-82
- use [tmdb-194777] to force the search to assign a TMDB GUID such as "tmdb-194777"
- update metadata and post the logs showing a TMDB GUID.

**Comment by RainOne** [comment]
Well, I did what you asked (i commented the lines you asked to delete): 
![image](https://user-images.githubusercontent.com/7484546/107255923-931ee680-6a17-11eb-9fe0-a9297f7a582b.png)
![image](https://user-images.githubusercontent.com/7484546/107255944-9a45f480-6a17-11eb-96ff-4cc6166533cc.png)

and i created a new database and made a new scan. The new scan only brought in the data from the TVDB forced entries. Looking at the logs, it seems that the TMDB is being assigned, but the scanner is not finding anything:

[Kamen Rider Shin (1992) [tmdb-194777].agent-update.log](https://github.com/ZeroQI/Hama.bundle/files/5945450/Kamen.Rider.Shin.1992.tmdb-194777.agent-update.log)
[Kamen Rider Shin (1992) [tmdb-194777].filelist.log](https://github.com/ZeroQI/Hama.bundle/files/5945451/Kamen.Rider.Shin.1992.tmdb-194777.filelist.log)
[Kamen Rider Shin (1992) [tmdb-194777].scanner.log](https://github.com/ZeroQI/Hama.bundle/files/5945452/Kamen.Rider.Shin.1992.tmdb-194777.scanner.log)

the id is being assigned, but the tmdb.getmetadata function is still not getting it.

**Comment by ZeroQI** [comment]
You are right, no TMDbid detected
```
=== TheMovieDb.GetMetadata() ================================================================================================================================
TVDBid: '', TMDbid: '', IMDbid: ''
```
I messed up
Put back deleted lines, use this line to replace (forgot an undercore for 2 variables)
`return AnimeLists_dict, AniDB_winner, TVDB_winner if TVDB_winner.isdigit() else "", Dict(mappingList, 'tmdbid') or TMDB_id, Dict(mappingList, 'imdbid') or IMDB_id, mappingList`

**Comment by RainOne** [comment]
So that worked, the logs now show that the TMDB is being assigned and OMDb (i already set up the API key) is also identifying the IMDB instance of the movie.

[Kamen Rider Shin (1992) [tmdb-194777].agent-search.log](https://github.com/ZeroQI/Hama.bundle/files/5947701/Kamen.Rider.Shin.1992.tmdb-194777.agent-search.log)
[Kamen Rider Shin (1992) [tmdb-194777].agent-update.log](https://github.com/ZeroQI/Hama.bundle/files/5947702/Kamen.Rider.Shin.1992.tmdb-194777.agent-update.log)


However, only the "genre" is being pulled from TMDB, and I imagine that's because TheMovieDB and OMDb are on the list of "T--M 'genres'" on HAMA configuration. I don't really understand how the service tag configurationn works (you said before that posters were not being pulled because of priorities, but as I understand, priorities only work when the request works - if it doesn't, it shouldn't prioritize "no answer"). It seems that addimg TheMovieDB to those different tags didn't work - unless it requires a full plex server reboot/cache clean.

I also noticed that the title is not being pulled on either (TMDB or OMDb), as per logs. Everything else is (description, posters, background images), but nothing is being applied on my library - I imagine this is a Cache problem as some data is already cached. 

I always have issues with Plex cached items and I don't know how to forcibly clean them other than deleting everything - and I don't really want to remove all the chaced information from my other libraries. 

Seems we're going the right way.

**Comment by ZeroQI** [comment]
**_If no title is set, no metadata is saved_**
- Please set title to 'AniDB, TheTVDB, TheMovieDb, OMDb | TheTVDB, AniDB' so it can get the title and save metadata
- [#] means no meta, but add the source if it says the field is provided (Inside: ['metadata source1', 'metadata source 2']) for summary, countries, originally_available_at in this instance...
- parenthesis around selected metadata source, but the metadata source order in settings is kept...

```
Metadata Fields (items #), type, source provider, value
[!] field: "title" not in meta_root, source: "None"
[#] originally_available_at        Sources: AniDB,TheTVDB                                                 Inside: ['OMDb']  Values: 
[!] 'MyAnimeList' source not in MetaSources dict, please Check case and spelling
[=] duration                       Sources: TheTVDB, AniDB, MyAnimeList, (TheMovieDb), OMDb               Inside: '['TheMovieDb', 'OMDb']'  Value: '88'
[!] 'MyAnimeList' source not in MetaSources dict, please Check case and spelling
[=] rating                         Sources: AniDB, TheTVDB, MyAnimeList, (TheMovieDb), OMDb | TheTVDB, AniDB  Inside: '['TheMovieDb', 'OMDb']'  Value: '4.4'
[!] 'MyAnimeList' source not in MetaSources dict, please Check case and spelling
[=] genres                   ( 3)  Sources: TheTVDB, AniDB, MyAnimeList, (TheMovieDb), OMDb               Inside: '['TheMovieDb', 'OMDb']'  Value: '['Action', 'Adventure', 'Science Fiction']'
[!] 'MyAnimeList' source not in MetaSources dict, please Check case and spelling
[#] summary                        Sources: TheTVDB, AniDB                                                Inside: ['TheMovieDb', 'OMDb']  Values: 
[#] countries                      Sources: TheMovieDb                                                    Inside: ['OMDb']  Values: 
[=] studio                         Sources: AnimeLists, AniDB, (TheMovieDb)                               Inside: '['TheMovieDb']'  Value: 'Toei Company'
[#] countries                      Sources: TheMovieDb                                                    Inside: ['OMDb']  Values: 
[?] meta_new: {
      'https://image.tmdb.org/t/p/original/2FRSGUsD571KxR7Rne60PKZ99EW.jpg': ('TheMovieDb/poster/2FRSGUsD571KxR7Rne60PKZ99EW.jpg', 17, None)}
    meta_old: [
      ]
[x] posters                  ( 1)  Sources: tvdb4, (TheMovieDb), TheTVDB, AniList, FanartTV, AniDB        Inside: '['TheMovieDb', 'OMDb']'  Value: '{'https://image.tmdb.org/t/p/original/2FRSGUsD571KxR7Rne60PKZ99EW.jpg': ('TheMovieDb/poster/2FRSGUsD571KxR7Rne60PKZ99EW.jpg', 17, None)}'
[x] art                      ( 1)  Sources: TheTVDB, (TheMovieDb), FanartTV                               Inside: '['TheMovieDb']'  Value: '{'https://image.tmdb.org/t/p/original/178rrQRHM4RnswUMXjC1JB6ilqU.jpg': ('TheMovieDb/artwork/178rrQRHM4RnswUMXjC1JB6ilqU.jpg', 34, 'https://image.tmdb.org/t/p/w300/178rrQRHM4RnswUMXjC1JB6ilqU.jpg')}'
```

**Comment by RainOne** [comment]
I added as you requested (I imagine it's the T-EM 'title"?)
![image](https://user-images.githubusercontent.com/7484546/107301278-f6316d00-6a59-11eb-9628-123bee971536.png)

Still no Title found on the logs: 
[Kamen Rider Shin (1992) [tmdb-194777].agent-search.log](https://github.com/ZeroQI/Hama.bundle/files/5947937/Kamen.Rider.Shin.1992.tmdb-194777.agent-search.log)
[Kamen Rider Shin (1992) [tmdb-194777].agent-update.log](https://github.com/ZeroQI/Hama.bundle/files/5947938/Kamen.Rider.Shin.1992.tmdb-194777.agent-update.log)

I also tried adding to the title_sort and original_title, but they also didn't provide a title in the logs.

**Comment by ZeroQI** [comment]
It is moving forward, metadata in update log looks great apart from the title which exists in 'ja' only but not found:
https://api.tmdb.org/3/movie/194777?api_key=7f4a0bd0bd3315bb832e17feda70b5cd&append_to_response=releases,credits,trailers,external_ids&language=en

language_series : ['en', 'ja'] is set so it needs code amended to get the 'ja' title from TMDB.

**Comment by RainOne** [comment]
I didn't really understand the problem: The api link you pasted does have both the "original title" and the "title", it's just way down below (after languages): 

"runtime":88,"spoken_languages":[{"english_name":"Japanese","iso_639_1":"ja","name":"日本語"}],"status":"Released","tagline":"","title":"Shin Kamen Rider: Prologue","video":false,"vote_average":4.4,"vote_count":5,"releases":{"countries":[{"certification":"G","iso_3166_1":"JP","primary":false,"release_date":"1992-02-20"},{"certification":"","iso_3166_1":"JP","primary":false,"release_date":"2005-12-24"}]},"credits":{"cast":[{"adult":false,"gender":2,"id":1175053,"known_for_department":"Acting","name":"Shin Ishikawa","original_name":"Shin Ishikawa","popularity":0.608,"profile_path":null,

I don't know if you mean "title from ja" as in the original title in japanese (which wouldn't be useful) or if i'm just misunderstanding you.

Probably the latter.

**Comment by ZeroQI** [comment]
i missed the `"title":"Shin Kamen Rider: Prologue"`
if a movie it's called 'title', if a series it's called 'name', and here it's a movie in a series library...

https://github.com/ZeroQI/Hama.bundle/blob/master/Contents/Code/TheMovieDb.py line 43 change
`    Log.Info("[ ] title: {}"                  .format(SaveDict( Dict(json, 'title' if mode=="movie" else 'name'), TheMovieDb_dict, 'title'                  )))`
into
`    Log.Info("[ ] title: {}"                  .format(SaveDict( Dict(json, 'title') or Dict(json, 'name'), TheMovieDb_dict, 'title'                  )))`

Submitted 2 fixed to include in master code

**Comment by RainOne** [comment]
yeah, that worked.

A couple more things: 

OMDb.py is looking for "title" on line 29, it should be "Title", as the OMDb api request calls it (with a capital T):
`{"Title":"Shin Kamen Rider","Year":"1992","Rated":"N/A","Released":"20 Feb 1992","Runtime":"90 min","Genre":"Action, Adventure, Drama, Horror, Sci-Fi, Thriller","Director":"Makoto Ishikawa, Makoto Tsuji","Writer":"Shotaro Ishinomori, Hayato Miyashita, Joe Onodera","Actors":"Kohisa Ishikawa, Yumi Nomura, Joe Onodera, Akira Ishihama","Plot":"In Shotaro Ishinomori's own deconstructionist entry in his famous \"Masked Rider\" series, young Shin Kazamatsuri is transformed into a powerful grasshopper mutant by a scientific institute known only as the \"Foundation,\" which targets him.","Language":"Japanese, English","Country":"Japan","Awards":"N/A","Poster":"https://m.media-amazon.com/images/M/MV5BOTZhNWM5YTEtYzNlNi00YzdlLWJkYmMtOGUzMzBhYjA5NWUyXkEyXkFqcGdeQXVyNjUwMTE5MjI@._V1_SX300.jpg","Ratings":[{"Source":"Internet Movie Database","Value":"5.7/10"}],"Metascore":"N/A","imdbRating":"5.7","imdbVotes":"113","imdbID":"tt0157028","Type":"movie","DVD":"N/A","BoxOffice":"N/A","Production":"N/A","Website":"N/A","Response":"True"}`


TMDb.py doesn't grab "originally_available_at" on line 49 because it's looking for "first_air_date", while TMDb offers that value as "release_date":"1992-02-20" (for movies, I imagine?). I added an "or dict(json, 'release_date')", on line 49 of TheMovieDb.py and that is also scrapped now.

I was gonna say that Summary and Year were not being added to the plex interface, but I noticed that TheMovieDB and OMDb were not listed on Hama as sources for that information, so I added them and now they show up. 

I'll keep looking if there's any more interesting things to add - how should I suggest them? Just paste them in here?
